contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,850 @@
+    // controller from the PodTemplateSpec indicated by CurrentTemplateRevision.
+ 	CurrentReplicas int32 `json:"currentReplicas,omitempty"`
+ 	
+ 	// TargetRevisionReplicas is the number of Pods created by the StatefulSet
+    // controller from the PodTemplateSpec indicated by TargetTemplateRevision.
+ 	TargeReplicas int32 `json:"taretReplicas,omitempty"`
+}
+```
+
+The following modifications will be made to the StatefulSetSpec API object.
+
+```go
+type StatefulSetSpec struct {
+    // Replicas, Selector, Template, VolumeClaimsTemplate, and ServiceName 
+    // ommitted for brevity.
+	v1.PodTemplateSpec `json:"template"`
+
+	// TemplateRevision is a monotonically increasing, 64 bit, integer used to 
+	// indicate the version of the of the PodTemplateSpec. If nil, the 
+	// StatefulSetController has not initialized its revision history,
+	// change tracking is not enabled, and all Pods will be created from 
+	// Template.
+	TemplateRevision  *int64 `json:"templateRevision"`
+
+	// RevisionParition paritions the Pods in the StatefulSet by ordinal such 
+    // that all Pods with a lower ordinal will be created from the PodTemplate that
+    // represents the current revision of the StatefulSet's revision history and 
+    // all Pods with an a greater or equal ordinal will be created from the 
+    // PodTemplate that represents the target revision of the StatefulSet's 
+    // revision history.
+	RevisionPartition *int32 `json:"revisionPartition,omitempty`
+
+	// RevisionHistoryDepth is the maximum number of PodTemplates that will 
+	// be maintained in the StatefulSet's revision history. It must be at 
+	// least two.
+	RevisionHisotryDepth int32 `json:historyRevisionDepth,omitempty`
+}
+```
+
+Additionally, we introduce the following constants.
+
+```go 
+// StatefulSetPodTemplateLabel is the label applied to a PodTemplate to allow
+// the StatefulSet controller to select the PodTemplates in its revision 
+// history.
+const StatefulSetPodTemplateLabel = "created-by-statefulset"
+
+// StatefulSetTemplateRevisionLabel is the label applied to a PodTemplate or 
+// Pod to indicate the position of the object's Template in the revision 
+// history of a StatefulSet.
+const StatefulSetTemplateRevisionLabel = "statefulset-template-revision"
+```
+
+## StatefulSet Controller
+The StatefulSet controller will watch for modifications to StatefulSet and Pod 
+API objects. When a StatefulSet is created or updated, or when one 
+of the Pods in a StatefulSet is updated or deleted, the StatefulSet
+controller will attempt to create, update, or delete Pods to conform the 
+current state of the system to the user declared target state. 
+The user declared target state of the system, with respect to an individual 
+StatefulSet, is determined as below.
+
+### Target State
+The declared target state of a StatefulSet requires that all Pods in the 
+StatefulSet conform to exactly one or two PodTemplates in the StatefulSet's 
```

Comment:
In essence we still will. The target revision of the StatefulSet's history will be semantically, deeply equivalent to its Spec.Template with the exception that it will carry the annotation for revision history that is applied to the PodTemplate. That is reason that the PodTemplate and not the Spec.Template will be used for creation.